subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/JULRedirector.java

```diff
@@ -37,14 +38,9 @@
     @Override
     public StandardOutputCapture start() {
         super.start();
-        boolean shouldReadLoggingConfigFile = System.getProperty(READ_LOGGING_CONFIG_FILE_PROPERTY, "true").equals("true");
-        if (!shouldReadLoggingConfigFile) {
-            SingleMessageLogger.nagUserOfDiscontinuedProperty(READ_LOGGING_CONFIG_FILE_PROPERTY,
-                "Change your test to work with your java.util.logging configuration file settings.");
-        }
         if (!reset) {
             LogManager.getLogManager().reset();
-            if (shouldReadLoggingConfigFile) {
+            if (shouldReadLoggingConfigFile()) {
```

Comment:
❌ Why was this changed?

Reply:
Before this change, deprecation check of `READ_LOGGING_CONFIG_FILE_PROPERTY` happened in test worker JVM, then the warning messages are forwarded to main JVM. It's impossible to collect warnings from another JVM, so we have to move this part to main JVM.